ruby1.rb

```diff
@@ -0,0 +1,258 @@
+ # Ruby HW
+
+# ![ARRAYS](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQVWBMdo6Ac3moY3tPnzMsFVnOscOR03SxkZ4sPGGhsWoQrYMPZ9g)
+
+## 1. Return an array of each Student's full name, upper-cased -->
+
+
+students = [
+  {
+      first_name: 'John',
+      last_name: 'Doe'
+  },
+  {
+      first_name: 'Jane',
+      last_name: 'Doe',
+  },
+  {
+      first_name: 'Jennifer',
+      last_name: 'Aniston',
+  }
+]
+
+
+  
+
+
+
+ upper_case_full_names=students.map do |i|
+    i[:first_name].upcase+" "+i[:last_name].upcase
+    end
+    puts upper_case_full_names
+
+
+
+
+# <!-- ### Answer
+
+# # ```rb
+
+# # [ 'JOHN DOE', 'JANE DOE', 'JENNIFER ANISTON' ] -->
+
+# ## 2. Find the first order for each user
+
+
+users = [
+  {
+      name: 'Ahmed',
+      orders: [
+          {
+              description: 'a bike'
+          }
+      ]
+  },
+  {
+      name: 'Sami',
+      orders: [
+          {
+              description: 'bees'
+          },
+          {
+              description: 'fishing rod'
+          }
+      ]
+  },
+  {
+      name: 'Salman',
+      orders: [
+          {
+              description: 'a MacBook'
+          },
+          {
+              description: 'The West Wing DVDs'
+          },
+          {
+              description: 'headphones'
+          },
+          {
+              description: 'a kitten'
+          }
+      ]
+  }
+]
+  arr=users[0][:orders][0],users[1][:orders][0],users[2][:orders][0]
+ puts arr
+#  //+[1][:orders]+[2][:orders]+[3][:orders]
+
+
+first_order_for_each_user =arr
+
+# ```
+
+# # ### Answer
+
+# # ```rb
+
+[ {description: "a bike"}, {description: "bees"}, {description: "a MacBook"} ]
+
+##3. Find the average amount spent on coffee, per transaction, for each person
+
+
+
+
+people = [
+  {
+      name: 'Maha',
+      transactions: [
+          {
+              type: 'COFFEE',
+              amount: 7.43
+          },
+          {
+              type: 'TACOS',
+              amount: 14.65
+          },
+          {
+              type: 'COFFEE',
+              amount: 4.43
+          }
+      ]
+  },
+  
+  {
+      name: 'Salman',
+      transactions: [
+          {
+              type: 'BIKES',
+              amount: 800.00
+          },
+          {
+              type: 'TACOS',
+              amount: 14.65
+          },
+          {
+              type: 'COFFEE',
+              amount: 4.43
+          }
+      ]
+  },
+  {
+      name: 'Sami',
+      transactions: [
+          {
+              type: 'COFFEE',
+              amount: 7.43
+          },
+          {
+              type: 'COFFEE',
+              amount: 100.00
+          },
+          {
+              type: 'COFFEE',
+              amount: 4.43
+          }
+      ]
+  }
+]
+
+
+coffee_average_per_person = people do |Array|
```

Comment:
```Array``` is a reserved word in ruby so we can not be utilized in the creation of variable names